src/main/java/org/reaktivity/nukleus/kafka/internal/stream/NetworkConnectionPool.java

```diff
@@ -641,7 +641,12 @@ private void handleData(
                 {
                     if (networkSlotOffset == 0 && networkSlot != NO_SLOT)
                     {
-                        if (networkSlot != LOCAL_SLOT)
+                        if (networkSlot == LOCAL_SLOT)
+                        {
+                            localDecodeCapacity = 0;
+                            localDecodeBuffer = null;
```

Comment:
Doesn't this now prevent local buffer reuse?

Reply:
my understanding is we never reuse the local buffer. Did I miss something?

Reply:
wrong! Fixed and pushed